src/Compilers/Core/Portable/CommandLine/SarifV2ErrorLogger.cs

```diff
@@ -351,41 +376,40 @@ private sealed class DiagnosticDescriptorSet
             /// <returns>
             /// The unique key assigned to the given descriptor.
             /// </returns>
-            public int Add(DiagnosticDescriptor descriptor, bool? hasAnyExternalSuppression = null)
+            public int Add(DiagnosticDescriptor descriptor, DiagnosticDescriptorErrorLoggerInfo? info = null)
             {
-                if (_distinctDescriptors.TryGetValue(descriptor, out var descriptorInfo))
+                if (_distinctDescriptors.TryGetValue(descriptor, out var descriptorInfoWithIndex))
                 {
                     // Descriptor has already been seen.
                     // Update 'HasAnyExternalSuppression' value if different from the saved one.
```

Comment:
Is this comment still correct? It looks like we're updating the `HasAnyExternalSuppression` and the timing state if either have changed.